sdk/helper/ldaputil/client.go

```diff
@@ -489,5 +489,14 @@ func getTLSConfig(cfg *ConfigEntry, host string) (*tls.Config, error) {
 		}
 		tlsConfig.RootCAs = caPool
 	}
+	if cfg.ClientTLSCert != "" && cfg.ClientTLSKey != "" {
+		certificate, err := tls.X509KeyPair([]byte(cfg.ClientTLSCert), []byte(cfg.ClientTLSKey))
+		if err != nil {
+			return nil, errwrap.Wrapf("failed to parse client X509 key pair: {{err}}", err)
+		}
+		tlsConfig.Certificates = append(tlsConfig.Certificates, certificate)
+	} else if (cfg.ClientTLSCert != "" && cfg.ClientTLSKey == "") || (cfg.ClientTLSCert == "" && cfg.ClientTLSKey != "") {
```

Comment:
```suggestion
	} else if cfg.ClientTLSCert != "" || cfg.ClientTLSKey != "") {
```

Reply:
That hits the case where you don't use either of them (which happens in case you use other type of authentication). Do we really wanna throw an error in that case?

Reply:
Why should it hit the case where you don't use either of them? It is only true if the client cert or the client key has been provided but not both.